app/src/main/java/eu/faircode/email/Core.java

```diff
@@ -1050,49 +1053,57 @@ private static void onSynchronizeMessages(
             if (keep_time < 0)
                 keep_time = 0;
 
-            Log.i(folder.name + " sync=" + new Date(sync_time) + " keep=" + new Date(keep_time));
-
-            // Delete old local messages
-            if (auto_delete) {
-                List<Long> tbds = db.message().getMessagesBefore(folder.id, keep_time, delete_unseen);
-                Log.i(folder.name + " local tbd=" + tbds.size());
-                EntityFolder trash = db.folder().getFolderByType(folder.account, EntityFolder.TRASH);
-                for (Long tbd : tbds) {
-                    EntityMessage message = db.message().getMessage(tbd);
-                    if (message != null && trash != null)
+            Date syncDate = new Date(sync_time);
+            if (folder.keep_days == Integer.MAX_VALUE) {
+                Log.i(folder.name + " sync=" + syncDate + " keep=∞");
+            } else {
+                Log.i(folder.name + " sync=" + syncDate + " keep=" + new Date(keep_time));
+
+                // Delete old local messages
+                if (auto_delete) {
+                    List<Long> tbds = db.message().getMessagesBefore(folder.id, keep_time, delete_unseen);
+                    Log.i(folder.name + " local tbd=" + tbds.size());
+                    EntityFolder trash = db.folder().getFolderByType(folder.account, EntityFolder.TRASH);
+                    for (Long tbd : tbds) {
+                        EntityMessage message = db.message().getMessage(tbd);
+                        if (message != null && trash != null)
                         if (EntityFolder.TRASH.equals(folder.type))
                             EntityOperation.queue(context, message, EntityOperation.DELETE);
                         else
                             EntityOperation.queue(context, message, EntityOperation.MOVE, trash.id);
+                    }
+                } else {
+                    int old = db.message().deleteMessagesBefore(folder.id, keep_time, delete_unseen);
+                    Log.i(folder.name + " local old=" + old);
                 }
-            } else {
-                int old = db.message().deleteMessagesBefore(folder.id, keep_time, delete_unseen);
-                Log.i(folder.name + " local old=" + old);
             }
 
             // Get list of local uids
             final List<Long> uids = db.message().getUids(folder.id, sync_kept ? null : sync_time);
             Log.i(folder.name + " local count=" + uids.size());
 
             // Reduce list of local uids
-            SearchTerm searchTerm = new ReceivedDateTerm(ComparisonTerm.GE, new Date(sync_time));
+            ReceivedDateTerm receivedDateSyncTerm = new ReceivedDateTerm(ComparisonTerm.GE, syncDate);
+            SearchTerm searchTerm = receivedDateSyncTerm;
```

Comment:
I don't think this is going to make a big difference

Reply:
Same